crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -485,6 +481,153 @@ impl DecisionTree {
+        for i in &candidates {
+            if intersect(&solution, i).is_none() {
+                solution.push(Segment { a: i.a, b: i.b });
+            }
+        }
+
+        // 3. Merge the blocks using the solution
+        let mut left_pos = 0;
+        let mut right_pos = 0;
+        let mut result = Vec::new();
+        for i in solution {
+            while left_pos < i.a.0 {
+                result.push(left[left_pos]);
+                left_pos += 1;
+            }
+            while right_pos < i.b.0 {
+                result.push(right[right_pos]);
+                right_pos += 1;
+            }
+            //merge i:
+            let ins = ctx.get_instruction(left[left_pos]);
+            let jins = ctx.get_instruction(right[right_pos]);
+            let assumption = &self[ctx[block_id].assumption];
+
+            let mut to_add = Vec::new();
+            let mut merged_op = match &ins.operation {
+                Operation::Call { arguments, .. } => {
+                    if let Operation::Call { func_id, arguments: a2, returned_arrays, .. } =
+                        &jins.operation
+                    {
+                        for a in arguments.iter().enumerate() {
+                            let op = Operation::Cond {
+                                condition: self[assumption.parent].condition,
+                                val_true: *a.1,
+                                val_false: a2[a.0],
+                            };
+                            let typ = ctx.get_object_type(*a.1);
+                            to_add.push(Instruction::new(op, typ, Some(block_id)));
+                        }
+                        Operation::Call {
+                            func_id: *func_id,
+                            arguments: Vec::new(),
+                            returned_arrays: returned_arrays.clone(),
+                            predicate: self.root,
+                        }
+                    } else {
+                        unreachable!();
+                    }
+                }
+                Operation::Store { array_id, index, value } => {
+                    if let Operation::Store { value: j_value, .. } = jins.operation {
+                        let op = Operation::Cond {
+                            condition: self[assumption.parent].condition,
+                            val_true: *value,
+                            val_false: j_value,
+                        };
+                        let merge =
+                            Instruction::new(op, ctx.mem[*array_id].element_type, Some(block_id));
+                        to_add.push(merge);
+                        // TODO: we should simplify this instruction
+                    }
+                    Operation::Store { array_id: *array_id, index: *index, value: NodeId::dummy() }
+                }
+                _ => unreachable!(),
+            };
+            let mut to_add_id = Vec::new();
```

Comment:
`ids_to_add`

(add to what?)
Another option is `merge_ids` which names them as what they are rather than what they'll be later used for.

Reply:
Done, I used to_merge and merge_ids